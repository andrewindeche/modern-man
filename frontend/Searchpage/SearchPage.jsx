import React from "react";
import NotificationBar from '../src/components/NotificationBar';
import SearchBar from '../src/components/SearchBar';
import NavButtons from '../src/components/NavButtons';
import tuxedo4 from '../src/images/tuxedo4.webp'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faStar} from '@fortawesome/free-solid-svg-icons'; 

const Searchpage = () => {
    return (
    <>
    <NotificationBar />
    <NavButtons />
    <SearchBar />
    <h4 className="searchresultstitle">Suits and Tuxedos</h4>
    <div className="searchresultsimages">
    <div className="image">
    <div className="container">
        <span className="ondiscount">-18%</span>
        <img src={tuxedo4} alt="Italian Suits" />
        <p>Black Italian Tuxedo</p>
        {[...Array(4)].map((_, index) => (
        <FontAwesomeIcon key={index} icon={faStar} className="shopping" />
    ))}
        <p id="price">$1000</p>
    </div>
    </div>
    <span className="tooltip-text">View More</span>
    <div className="image">
    <div className="container">
    <span className="onsale">New</span>
        <img src={tuxedo4} alt="Grey Suit" />
        <p>Grey Official Suit</p>
        {[...Array(4)].map((_, index) => (
        <FontAwesomeIcon key={index} icon={faStar} className="shopping" />
    ))}
        <p id="price">$700</p>
        </div>
        </div>
        <span className="tooltip-text">View More</span>
    <div className="image">
        <div className="container">
        <span className="ondiscount">-18%</span>
            <img src={tuxedo4} alt="Blue Tuxedo" />
        <p>Blue Official Tuxedo</p>
        {[...Array(4)].map((_, index) => (
        <FontAwesomeIcon key={index} icon={faStar} className="shopping" />
    ))}
    <p id="price">$1000</p>
    </div>
    </div>
    <span className="tooltip-text">View More</span>
    <div className="image">
    <div className="container">
      <span className="onsale">New</span>
        <img src={tuxedo4} alt="White Suit" />
        <p>White Official Suit</p>
        {[...Array(4)].map((_, index) => (
        <FontAwesomeIcon key={index} icon={faStar} className="shopping" />
    ))}
    <p id="price">$700</p>
    </div>
    </div>
    <span className="tooltip-text">View More</span>
</div>
    </>
    )         
}

export default Searchpage;