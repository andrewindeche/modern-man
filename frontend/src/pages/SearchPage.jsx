import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useLocation } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faStar } from '@fortawesome/free-solid-svg-icons';
import { fetchProducts } from '../store/productsSlice';
import { fetchDiscountedProducts } from '../store/discountsSlice';
import { searchProducts, updateQuery } from '../store/searchSlice';
import NotificationBar from '../components/NotificationBar';
import SearchBar from '../components/SearchBar';
import NavButtons from '../components/NavButtons';
import ModalContent from '../components/Modal';

const useQuery = () => new URLSearchParams(useLocation().search);

const SearchPage = () => {
  const dispatch = useDispatch();
  const products = useSelector((state) => state.products);
  const discount = useSelector((state) => state.discount);
  const search = useSelector((state) => state.search);
  const query = useQuery();
  const discounted = query.get('discounted') === 'true';
  const category = query.get('category') || '';
  const searchQuery = query.get('query') || '';
  const [selectedItem, setSelectedItem] = useState(null);
  const [modalPosition, setModalPosition] = useState({ top: 0, left: 0 });

  useEffect(() => {
    if (searchQuery) {
      dispatch(updateQuery(searchQuery));
      dispatch(searchProducts(searchQuery));
    } else if (discounted) {
      dispatch(fetchDiscountedProducts());
    } else {
      dispatch(fetchProducts({ category, discounted }));
    }
  }, [dispatch, searchQuery, category, discounted]);

  const { results: searchItems, loading: searchLoading, error: searchError } = search;
  const { items = [], loading, error } = discounted ? discount : products;
  const handleItemClick = (item, event) => {
    const rect = event.target.getBoundingClientRect();
    setModalPosition({ top: rect.top + window.scrollY, left: rect.left + window.scrollX });
    setSelectedItem(item);
  };
  const renderItems = (items) => (
    items.map((item, index) => (
      <div key={index} className="image" onClick={(event) => handleItemClick(item, event)}>
        <div className="container">
          {item.discount_percentage > 0 && (
          <span className="ondiscount">
            {item.discount_percentage}
            %
          </span>
          )}
          <img src={item.image} alt={item.name} />
          <p className="itemname">{item.name}</p>
          {[...Array(5)].map((_, i) => (
            <FontAwesomeIcon
              // eslint-disable-next-line react/no-array-index-key
              key={i}
              icon={faStar}
              className={`star ${i < item.average_rating ? 'active' : ''}`}
              size="2x"
            />
          ))}
          <p id="price">
            $
            {item.discounted_price}
          </p>
        </div>
      </div>
    ))
  );

  return (
    <>
      <NotificationBar />
      <NavButtons />
      <SearchBar />
      <div className="searchresultsimages">
        {searchQuery ? (
          searchLoading ? (
            <p>Loading...</p>
          ) : searchError ? (
            <p>
              Error:
              {searchError}
            </p>
          ) : (
            renderItems(searchItems)
          )
        ) : (
          loading ? (
            <p>Loading...</p>
          ) : error ? (
            <p>
              Error:
              {error}
            </p>
          ) : (
            renderItems(items)
          )
        )}
        <span className="tooltip-text">View More</span>
      </div>
      {selectedItem && (
        <div className="modal-background" onClick={() => setSelectedItem(null)}>
          <ModalContent item={selectedItem} onClose={() => setSelectedItem(null)} position={modalPosition} />
        </div>
      )}
    </>
  );
};

export default SearchPage;
