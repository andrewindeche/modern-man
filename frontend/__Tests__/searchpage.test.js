import React from 'react';
import SearchPage from 'pages/SearchPage.jsx';
import renderer from 'react-test-renderer';
describe("search page",() =>{
  it('renders the landing page', () => {
    const tree = renderer.create(<SearchPage />).toJSON();
    expect(tree).toMatchSnapshot();
  })
})
