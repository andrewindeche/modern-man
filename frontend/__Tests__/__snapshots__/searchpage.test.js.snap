// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search page renders the landing page 1`] = `
[
  <div
    className="notificationbar"
  >
    <img
      alt="My Shoes"
      className="shoes"
      src=""
    />
    <p>
      Get 50% off Selected Shoes
    </p>
    <button
      id="notificationbutton"
    >
      Learn More
    </button>
  </div>,
  <div
    className="categories"
  >
    <button
      className="suits-image"
      style={
        {
          "backgroundImage": "url()",
        }
      }
    >
      Suits
    </button>
    <button
      className="shirts-image"
      style={
        {
          "backgroundImage": "url()",
        }
      }
    >
      Shirts
    </button>
    <button
      className="neck-wear"
      style={
        {
          "backgroundImage": "url()",
        }
      }
    >
      Neck wear & Accessories
    </button>
    <button
      className="shoes-image"
      style={
        {
          "backgroundImage": "url()",
        }
      }
    >
      Shoes
    </button>
  </div>,
  <div
    className="searchbar"
  >
    <h3>
      Modern Man
    </h3>
    <form
      className="search-form"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-magnifying-glass search-icon"
        data-icon="magnifying-glass"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      <input
        className="search-input"
        placeholder="Search for Men's Wear and Accessories"
        type="text"
      />
    </form>
    <div
      className="user-icons"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-user user"
        data-icon="user"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224 256A128 128 0 1 0 224 0a128 128 0 1 0 0 256zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512H418.3c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304H178.3z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      <span
        className="tooltip-text"
      >
        User Profile
      </span>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-cart-shopping shopping"
        data-icon="cart-shopping"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 24C0 10.7 10.7 0 24 0H69.5c22 0 41.5 12.8 50.6 32h411c26.3 0 45.5 25 38.6 50.4l-41 152.3c-8.5 31.4-37 53.3-69.5 53.3H170.7l5.4 28.5c2.2 11.3 12.1 19.5 23.6 19.5H488c13.3 0 24 10.7 24 24s-10.7 24-24 24H199.7c-34.6 0-64.3-24.6-70.7-58.5L77.4 54.5c-.7-3.8-4-6.5-7.9-6.5H24C10.7 48 0 37.3 0 24zM128 464a48 48 0 1 1 96 0 48 48 0 1 1 -96 0zm336-48a48 48 0 1 1 0 96 48 48 0 1 1 0-96z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      <span
        className="tooltip-text"
      >
        Checkout
      </span>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-heart heart"
        data-icon="heart"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M47.6 300.4L228.3 469.1c7.5 7 17.4 10.9 27.7 10.9s20.2-3.9 27.7-10.9L464.4 300.4c30.4-28.3 47.6-68 47.6-109.5v-5.8c0-69.9-50.5-129.5-119.4-141C347 36.5 300.6 51.4 268 84L256 96 244 84c-32.6-32.6-79-47.5-124.6-39.9C50.5 55.6 0 115.2 0 185.1v5.8c0 41.5 17.2 81.2 47.6 109.5z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      <span
        className="tooltip-text"
      >
        Favorite
      </span>
    </div>
  </div>,
  <h4
    className="searchresultstitle"
  >
    Suits and Tuxedos
  </h4>,
  <div
    className="searchresultsimages"
  >
    <div
      className="image"
    >
      <div
        className="container"
      >
        <span
          className="ondiscount"
        >
          -18%
        </span>
        <img
          alt="Italian Suits"
          src=""
        />
        <p>
          Black Italian Tuxedo
        </p>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-star shopping"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={{}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
            style={{}}
          />
        </svg>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-star shopping"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={{}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
            style={{}}
          />
        </svg>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-star shopping"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={{}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
            style={{}}
          />
        </svg>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-star shopping"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={{}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
            style={{}}
          />
        </svg>
        <p
          id="price"
        >
          $1000
        </p>
      </div>
    </div>
    <span
      className="tooltip-text"
    >
      View More
    </span>
    <div
      className="image"
    >
      <div
        className="container"
      >
        <span
          className="onsale"
        >
          New
        </span>
        <img
          alt="Grey Suit"
          src=""
        />
        <p>
          Grey Official Suit
        </p>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-star shopping"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={{}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
            style={{}}
          />
        </svg>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-star shopping"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={{}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
            style={{}}
          />
        </svg>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-star shopping"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={{}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
            style={{}}
          />
        </svg>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-star shopping"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={{}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
            style={{}}
          />
        </svg>
        <p
          id="price"
        >
          $700
        </p>
      </div>
    </div>
    <span
      className="tooltip-text"
    >
      View More
    </span>
    <div
      className="image"
    >
      <div
        className="container"
      >
        <span
          className="ondiscount"
        >
          -18%
        </span>
        <img
          alt="Blue Tuxedo"
          src=""
        />
        <p>
          Blue Official Tuxedo
        </p>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-star shopping"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={{}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
            style={{}}
          />
        </svg>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-star shopping"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={{}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
            style={{}}
          />
        </svg>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-star shopping"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={{}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
            style={{}}
          />
        </svg>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-star shopping"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={{}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
            style={{}}
          />
        </svg>
        <p
          id="price"
        >
          $1000
        </p>
      </div>
    </div>
    <span
      className="tooltip-text"
    >
      View More
    </span>
    <div
      className="image"
    >
      <div
        className="container"
      >
        <span
          className="onsale"
        >
          New
        </span>
        <img
          alt="White Suit"
          src=""
        />
        <p>
          White Official Suit
        </p>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-star shopping"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={{}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
            style={{}}
          />
        </svg>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-star shopping"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={{}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
            style={{}}
          />
        </svg>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-star shopping"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={{}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
            style={{}}
          />
        </svg>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-star shopping"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={{}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
            fill="currentColor"
            style={{}}
          />
        </svg>
        <p
          id="price"
        >
          $700
        </p>
      </div>
    </div>
    <span
      className="tooltip-text"
    >
      View More
    </span>
  </div>,
]
`;
